crates/astria-grpc-mock/src/mock_server.rs

```diff
@@ -110,7 +110,7 @@ pub struct MockGuard {
 }
 
 impl MockGuard {
-    pub async fn wait_until_satisfied(self) {
+    pub async fn wait_until_satisfied(&self) {
```

Comment:
What makes it necessary to take this by reference? The timeout?

I actually thought of it as a feature that it gets dropped at the end of the scope, immediately returning the panic/report. But I suppose that wasn't useful and/or blocking some other functionality?

Reply:
If we had e.g.
```rust
tokio::time::timeout(duration, guard.wait_until_satisfied()).expect("shouldn't timeout");
```
then what was happening if `wait_until_satisfied` exceeded the duration was:
- `tokio::time::timeout` expired and cancelled the `wait_until_satisfied` future
- `wait_until_satisfied` had consumed the `MockGuard`, so when it was cancelled, the guard was dropped
- the `Drop` impl of the guard panicked due to unsatisfied condition
- we never get as far as executing `.expect("shouldn't timeout")` since we've panicked

I found this unexpected initially - I was expecting to see "shouldn't timeout" in the panic message, and then the report about the unsatisfied mock.  That what happens now by taking `&self` in `wait_until_satisfied`.